File: flutter_aws_amplify_cognito/android/src/main/kotlin/in/agnostech/flutter_aws_amplify_cognito/CognitoMethodChannelHandler.kt
package `in`.agnostech.flutter_aws_amplify_cognito

import android.content.Context
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel

class CognitoMethodChannelHandler(private val context: Context) : MethodChannel.MethodCallHandler {
    override fun onMethodCall(call: MethodCall, result: MethodChannel.Result) {
        when (call.method) {
            "isSignedIn" -> result.success(FlutterAwsAmplifyCognito.isSignedIn())
            "currentUserState" -> result.success(FlutterAwsAmplifyCognito.currentUserState())
            "initialize" -> FlutterAwsAmplifyCognito.initialize(context, result)
            "signOut" -> FlutterAwsAmplifyCognito.signOut()
            "signOutGlobally" -> FlutterAwsAmplifyCognito.signOutGlobally(result)
            "getUserAttributes" -> FlutterAwsAmplifyCognito.getUserAttributes(result)
            "getUsername" -> result.success(FlutterAwsAmplifyCognito.getUsername())
            "getIdentityId" -> result.success(FlutterAwsAmplifyCognito.getIdentityId())
            "getTokens" -> FlutterAwsAmplifyCognito.getTokens(result)
            "getIdToken" -> result.success(FlutterAwsAmplifyCognito.getIdToken())
            "getAccessToken" -> result.success(FlutterAwsAmplifyCognito.getAccessToken())
            "getRefreshToken" -> result.success(FlutterAwsAmplifyCognito.getRefreshToken())
            "getCredentials" -> FlutterAwsAmplifyCognito.getCredentials(result)
            "trackDevice" -> FlutterAwsAmplifyCognito.trackDevice(result)
            "untrackDevice" -> FlutterAwsAmplifyCognito.untrackDevice(result)
            "forgetDevice" -> FlutterAwsAmplifyCognito.forgetDevice(result)
            "getDeviceDetails" -> FlutterAwsAmplifyCognito.getDeviceDetails(result)
            "signUp" -> FlutterAwsAmplifyCognito.signUp(
                    result, call.argument<String>("username")!!,
                    call.argument<String>("password")!!,
                    call.argument<Map<String, String>>("userAttributes")!!
            )
            "confirmSignUp" -> FlutterAwsAmplifyCognito.confirmSignUp(
                    result,
                    call.argument<String>("username")!!,
                    call.argument<String>("code")!!
            )
            "resendSignUp" -> FlutterAwsAmplifyCognito.resendSignUp(
                    result,
                    call.argument<String>("username")!!
            )
            "signIn" -> FlutterAwsAmplifyCognito.signIn(
                    result,
                    call.argument<String>("username")!!,
                    call.argument<String>("password")!!
            )
            "confirmSignIn" -> FlutterAwsAmplifyCognito.confirmSignIn(
                    result,
                    call.argument<String>("confirmSignInChallenge")!!
            )
            "forgotPassword" -> FlutterAwsAmplifyCognito.forgotPassword(
                    result,
                    call.argument<String>("username")!!
            )
            "confirmForgotPassword" -> FlutterAwsAmplifyCognito.confirmForgotPassword(
                    result,
                    call.argument<String>("newPassword")!!,
                    call.argument<String>("confirmationCode")!!

            )
            "federatedSignIn" -> FlutterAwsAmplifyCognito.federatedSignIn(
                    result,
                    call.argument<String>("identityProvider")!!,
                    call.argument<String>("token")!!,
                    call.argument<String>("customRoleARN"),
                    call.argument<String>("cognitoIdentityId")
            )
            else -> result.notImplemented()
        }
    }

}

File: flutter_aws_amplify_cognito/ios/Classes/CognitoMethodChannelHandler.swift
//
//  CognitoMethodChannelHandler.swift
//  flutter_aws_amplify_cognito
//
//  Created by Vishal Dubey on 09/03/20.
//

import Foundation
import Flutter

func handleMethodCall(call: FlutterMethodCall, result: @escaping FlutterResult) {
    guard let arguments: [String: Any] = call.arguments as? [String : Any] else {
        result("Could not unwrap Flutter arguments")
        return
    }
    switch call.method {
    case "isSignedIn":
        result(SwiftFlutterAwsAmplifyCognito.isSignedIn())
        
    case "currentUserState":
        result(SwiftFlutterAwsAmplifyCognito.currentUserState())
        
    case "initialize":
        SwiftFlutterAwsAmplifyCognito.initialize(result: result)
        
    case "signOut":
        SwiftFlutterAwsAmplifyCognito.signOut()
        
    case "signOutGlobally":
        SwiftFlutterAwsAmplifyCognito.signOutGlobally(result: result)
        
    case "getUserAttributes":
        SwiftFlutterAwsAmplifyCognito.getUserAttributes(result: result)
        
    case "getUsername":
        result(SwiftFlutterAwsAmplifyCognito.getUsername())
        
    case "getIdentityId":
        result(SwiftFlutterAwsAmplifyCognito.getIdentityId())
        
    case "getTokens":
        SwiftFlutterAwsAmplifyCognito.getTokens(result: result)
        
    case "getIdToken":
        SwiftFlutterAwsAmplifyCognito.getIdToken(result: result)
        
    case "getAccessToken":
        SwiftFlutterAwsAmplifyCognito.getAccesstoken(result: result)
        
    case "getRefreshToken":
        SwiftFlutterAwsAmplifyCognito.getRefreshToken(result: result)
        
    case "getCredentials":
        SwiftFlutterAwsAmplifyCognito.getCredentials(result: result)
        
    case "trackDevice":
        SwiftFlutterAwsAmplifyCognito.trackDevice(result: result)
        
    case "untrackDevice":
        SwiftFlutterAwsAmplifyCognito.untrackDevice(result: result)
        
    case "forgetDevice":
        SwiftFlutterAwsAmplifyCognito.forgetDevice(result: result)
        
    case "getDeviceDetails":
        SwiftFlutterAwsAmplifyCognito.getDeviceDetails(result: result)
        
    case "signUp":
        SwiftFlutterAwsAmplifyCognito.signUp(result: result, username: arguments["username"] as! String, password: arguments["password"] as! String, userAttributes: arguments["userAttributes"] as! [String : String])
        
    case "confirmSignUp":
        SwiftFlutterAwsAmplifyCognito.confirmSignUp(result: result, username: arguments["username"] as! String, code: arguments["code"] as! String)
        
    case "resendSignUp":
        SwiftFlutterAwsAmplifyCognito.resendSignUp(result: result, username: arguments["username"] as! String)
        
    case "signIn":
        SwiftFlutterAwsAmplifyCognito.signIn(result: result, username: arguments["username"] as! String, password: arguments["password"] as! String)
        
    case "confirmSignIn":
        SwiftFlutterAwsAmplifyCognito.confirmSignIn(result: result, confirmSignInChallenge: arguments["confirmSignInChallenge"] as! String)
        
    case "forgotPassword":
        SwiftFlutterAwsAmplifyCognito.forgotPassword(result: result, username: arguments["username"] as! String)
        
    case "confirmForgotPassword":
        SwiftFlutterAwsAmplifyCognito.confirmForgotPassword(result: result, username: arguments["username"] as! String, newPassword: arguments["newPassword"] as! String, confirmationCode: arguments["confirmationCode"] as! String)
        
    default:
        result(FlutterMethodNotImplemented)
    }
}
